application/templates/_helpers.tpl

```diff
@@ -12,7 +12,7 @@ app: {{ template "application.name" . }}
 
 {{- define "application.labels.stakater" -}}
 {{ template "application.labels.selector" . }}
-appVersion: "{{ .Values.deployment.image.tag }}"
+appVersion: "{{ .Values.deployment.image.tag | trunc 63 | trimSuffix "-" -}}"
```

Comment:
why truncate at 63 chars?

Reply:
because some Kubernetes name fields are limited to this.